### PR TITLE
Use enum for UserDefaults keys, register defaults automatically

### PR DESCRIPTION
**Pull request by IMcD23**
Found a fix for #77. UserDefaults' defaults were getting registered after the root view controller was getting initialized in the AppDelegate.

This fix makes the UserDefaults register its default values on first access.

As part of that fix, I also took it upon myself to improve UserDefaults syntax across the app, by using a `Key` enum and subscripts.

### CODE REVIEW

File: OpenTerm/View/TerminalTextView.swift [hunk 1]
```diff
@@ -49,18 +49,17 @@ class TerminalTextView: UITextView {
 	}
 
 	private func updateAppearanceFromSettings() {
-		let userDefaultsController = UserDefaultsController.shared
+		let userDefaults = UserDefaults.terminalDefaults
 
-		let terminalFontSize = userDefaultsController.terminalFontSize
-		self.font = UIFont(name: "Menlo", size: CGFloat(terminalFontSize))
+		let fontSize: Int = userDefaults[.terminalFontSize]
```
Comment (louisdh) [inline OpenTerm/View/TerminalTextView.swift]: Hmm, I get the feeling that this enum approach is less type safe. One could accidentally try to assign to a wrong type, which would trigger a runtime crash. It's a great example of what's possible with Swift enums, but I believe it's the wrong abstraction for user defaults.

**Comment by IMcD23** [comment]
Good point. I will re-add the public getters/setters and keep the Keys enum private.

Would that resolve all issues you have with it’s current implementation?

**Comment by louisdh** [comment]
Yes, that'd be perfect.

**Comment by louisdh** [comment]
@IMcD23  Friendly reminder that this is still open. 🙂

**Comment by louisdh** [comment]
The UserDefaultsController has been updated since this PR was opened. Most recently with 08e15d3f7445c878a13daeaed3f75e5f1b9e921e, which no longer relies on `userDefaults.register(defaults)`. So I'm gonna close this, though you're welcome to revisit the enum approach in a new PR.

**Pull request closed without merging**